components/post_view/post/index.js

```diff
@@ -40,8 +44,21 @@ function makeMapStateToProps() {
     const createAriaLabelForPost = makeCreateAriaLabelForPost();
 
     return (state, ownProps) => {
+        const currentUserId = getCurrentUserId(state);
+        const currentTeamId = getCurrentTeamId(state);
         const post = ownProps.post || getPost(state, ownProps.postId);
         const channel = getChannel(state, post.channel_id);
+        const replyCount = getReplyCount(state, post) || post.reply_count || 0;
+        const collapsedThreads = isCollapsedThreadsEnabled(state);
+
+        let thread;
+        if (collapsedThreads && !post.root_id && replyCount) {
+            thread = getThreadOrSynthetic(state, post);
+
+            if (thread.is_following) {
+                fetchThread(currentUserId, currentTeamId, post.id);
```

Comment:
If the thread is a long one, would this affect the performance?

It also seems that all the info that is needed is used by the footer, so wouldn't it be better if we just pass the id to the footer and have that get the thread?

This will also prevent aditional thread fetches as the thread id won't change (only the first time that it goes from being a single post to being a thread) so there would only be one fetch regardless of any state change affecting the post.

Reply:
The thread here is just the `UserThread`, not the posts in it, so the performance would be consistent. But that's a good point to move it to ThreadFooter.. I'll do that.